src/overlays/actors/ovl_En_Fz/z_en_fz.c

```diff
@@ -26,73 +70,812 @@ const ActorInit En_Fz_InitVars = {
+}
+
+void EnFz_GetCollision(EnFz* this, GlobalContext* globalCtx) {
+    Vec3f pos;
+    Vec3f collisionPos;
+    Vec3f vec1;
+    u32 bgId;
+    CollisionPoly* poly;
+
+    pos.x = this->actor.posRot.pos.x;
+    pos.y = this->actor.posRot.pos.y + 20.0f;
+    pos.z = this->actor.posRot.pos.z;
+    Matrix_Translate(pos.x, pos.y, pos.z, MTXMODE_NEW);
+    Matrix_RotateRPY(this->actor.shape.rot.x, this->actor.shape.rot.y, this->actor.shape.rot.z, MTXMODE_APPLY);
+
+    vec1.y = 0.0f;
+    vec1.x = 0.0f;
+    vec1.z = 220.0f;
+    Matrix_MultVec3f(&vec1, &this->collisionPos);
+
+    // If Collided
+    if (func_8003DE84(&globalCtx->colCtx, &pos, &this->collisionPos, &collisionPos, &poly, 1, 0, 0, 1, &bgId)) {
+        Math_Vec3f_Copy(&this->collisionPos, &collisionPos);
+    }
+
+    pos.x = this->actor.posRot.pos.x - this->collisionPos.x;
+    pos.z = this->actor.posRot.pos.z - this->collisionPos.z;
+    this->xzDistSquare = SQ(pos.x) + SQ(pos.z);
+}
+
+s32 EnFz_IsCollision(EnFz* this, Vec3f* vec) {
+    f32 x_dist;
+    f32 z_dist;
+
+    x_dist = this->actor.posRot.pos.x - vec->x;
+    z_dist = this->actor.posRot.pos.z - vec->z;
+
+    if (this->xzDistSquare <= (SQ(x_dist) + SQ(z_dist))) {
+        return true;
+    }
+    else {
+        return false;
+    }
+}
+
+void EnFz_Damaged(EnFz* this, GlobalContext* globalCtx, Vec3f* vec, s32 numEffects, f32 unk_float) {
+    int i;
+    Vec3f pos;
+    Vec3f vel;
+    Vec3f accel;
+    Color_RGBA8 primColor;
+    Color_RGBA8 envColor;
+    f32 scale;
+    s32 life;
+
+    primColor.r = 155;
+    primColor.g = 255;
+    primColor.b = 255;
+    primColor.a = 255;
+    envColor.r = 200;
+    envColor.g = 200;
+    envColor.b = 200;
+    accel.z = 0.0f;
+    accel.y = -1.0f;
+    accel.x = 0.0f;
+
+    for (i = 0; i < numEffects; i++) {
+        scale = Math_Rand_CenteredFloat(0.3f) + 0.6f;
+        life = (s32)Math_Rand_CenteredFloat(5.0f) + 12;
+        pos.x = Math_Rand_CenteredFloat(unk_float) + vec->x;
+        pos.y = Math_Rand_ZeroFloat(unk_float) + vec->y;
+        pos.z = Math_Rand_CenteredFloat(unk_float) + vec->z;
+        vel.x = Math_Rand_CenteredFloat(10.0f);
+        vel.y = Math_Rand_ZeroFloat(10.0f) + 2.0f;
+        vel.z = Math_Rand_CenteredFloat(10.0f);
+        EffectSsEnIce_Spawn(globalCtx, &pos, scale, &vel, &accel, &primColor, &envColor, life);
+    }
+
+    func_80062CD4(globalCtx, vec);
+}
+
+void EnFz_SpawnIceSmokeHiddenState(EnFz* this) {
+}
+
+// Fully grown
+void EnFz_SpawnIceSmokeGrowingState(EnFz* this) {
+    Vec3f pos;
+    Vec3f velocity;
+    Vec3f accel;
+
+    if ((this->counter % 16) == 0) {
+        pos.x = Math_Rand_CenteredFloat(40.0f) + this->actor.posRot.pos.x;
+        pos.y = Math_Rand_CenteredFloat(40.0f) + this->actor.posRot.pos.y + 30.0f;
+        pos.z = Math_Rand_CenteredFloat(40.0f) + this->actor.posRot.pos.z;
+        accel.z = 0.0f;
+        accel.x = 0.0f;
+        velocity.z = 0.0f;
+        velocity.y = 0.0f;
+        velocity.x = 0.0f;
```

Comment:
same as before about chained assignment. Unclear if these are all a single chain or one chain for each vector. Whichever matches or looks best to you.

Reply:
That chain assignment makes so much sense. Velocity still needs to be below accel to match unfortunately, but chaining makes it look much cleaner